frontend/appflowy_flutter/lib/util/expand_views.dart

```diff
@@ -0,0 +1,27 @@
+import 'package:appflowy/workspace/application/view/view_bloc.dart';
+
+/// the key is view id
+final Map<String, Set<ViewBloc>> _recordBlocs = {};
+
+bool isViewExpanded(String id) => getViewBloc(id)?.state.isExpanded ?? false;
+
+void recordBloc(String id, ViewBloc bloc) {
+  final blocSet = _recordBlocs[id] ?? {};
+  blocSet.add(bloc);
+  _recordBlocs[id] = blocSet;
+}
+
+void eraseBloc(String id, ViewBloc bloc) {
+  final blocSet = _recordBlocs[id] ?? {};
+  blocSet.remove(bloc);
+  if (blocSet.isEmpty) {
+    _recordBlocs.remove(id);
+  } else {
+    _recordBlocs[id] = blocSet;
+  }
+}
+
+ViewBloc? getViewBloc(String id) {
+  final blocSet = _recordBlocs[id] ?? {};
+  return blocSet.isEmpty ? null : blocSet.first;
+}
```

Comment:
Using a global map to store the view bloc is not a suitable solution in this case. In my opinion, there are two better options:

1. Create a new bloc at the top level of the document and sidebar. Trigger the expand event in this new bloc when the view in the sidebar needs to expand. The sidebar can listen for updates from the bloc using `BlocListener`.
2. If using a global map, create a class to wrap these functions and provide a "namespace." Register this class in GetIt to allow for proper cleanup. Additionally, the names `recordBloc` and `eraseBloc` are too generic and difficult to understand in context.

Reply:
After trying out using Bloc, I found that Bloc does not seem to provide a way to listen to events outside of the Bloc.
It can only listen for state changes, although using state could work,  I believe it is not suitable for handling this issue. 
The best solution would be to use a top-level Bloc to manage expansion and collapse data, but that would require too many changes. 
So, I went with the second option and made the suggested changes.